server/somesockets.py

```diff
@@ -1,47 +1,81 @@
-from flask import Flask, render_template
+from flask import Flask, render_template, request
 from flask_socketio import SocketIO, send, emit
+from secrets import SOCKETIO_SECRET_KEY
+from models import Session
+from flask_mongoengine import MongoEngine
 
+#=====================================================
+# Global vars
+#=====================================================
 app = Flask(__name__)
-app.config['SECRET_KEY'] = 'secret!'
+app.config['SECRET_KEY'] = SOCKETIO_SECRET_KEY
+app.config.from_object('config')
+
+db = MongoEngine(app)
+
 socketio = SocketIO(app)
+open_rooms = dict()
+bijection = dict()
 
 #=====================================================
 # SocketIO
 #=====================================================
-
-socketio = SocketIO(app)
-
 @socketio.on('connect')
 def handle_connect():
-    print("a user connected")
+    room = request.query_string.decode()
+    if room == "":
+        send('Malformed request')
+        return
+
+    session = Session.objects(code=room).first()
+    if session is None:
+        send('Room not found')
+        return
+
+    open_rooms[room] += [request.sid]
+    bijection[request.sid] = room
+    emit('Connect', dict(message="Another user connected"), room=room)
+
 
 @socketio.on('disconnect')
 def handle_disconnect():
-    print("a user has left")
+    room = bijection[request.sid]
+    open_rooms[room].remove(request.sid)
```

Comment:
Don't we only want to remove the room from the open_rooms list if the last user has left?

Reply:
that's not how it's working, i'm storing all the users in a list and last line removes them from the list

Reply:
Ah, I see it now